src/core/qgstemporalnavigationobject.cpp

```diff
@@ -90,6 +90,9 @@ QgsDateTimeRange QgsTemporalNavigationObject::dateTimeRangeForFrameNumber( long
   const QDateTime begin = start.addSecs( frame * mFrameDuration.seconds() );
   const QDateTime end = start.addSecs( nextFrame * mFrameDuration.seconds() );
 
+  if ( mCumulativeTemporalRange )
+    return QgsDateTimeRange( start, begin, true, false );
```

Comment:
```suggestion
    return QgsDateTimeRange( start, end, true, false );
```

Reply:
This will cause the last frame date time range to go out of the animation temporal extent bounds.

Reply:
Could you apply the end time clamping first before doing this check then?

Reply:
Sure @nyalldawson, I'm thinking of updating the logic to look like this

```c++
const QDateTime begin = start.addSecs( frame * mFrameDuration.seconds() );
const QDateTime end = start.addSecs( nextFrame * mFrameDuration.seconds() );
  
QDateTime frameStart = begin;
 if ( mCumulativeTemporalRange )
     frameStart = start;

 if ( end <= mTemporalExtents.end() )
      return QgsDateTimeRange( frameStart, end, true, false );

return QgsDateTimeRange( frameStart, mTemporalExtents.end(), true, false );
```

are you okay with this?

Reply:
Yes, looks good!

Reply:
Okay